### DISCUSSION TITLE
Broaden the scope of HVAC_MODE_AUTO or add HVAC_MODE_ON

### DISCUSSION THREAD

**Discussion by Bre77** [home-assistant/architecture#625] [Entity Models]
In some home climate control systems, you have control of a master temperature and HVAC mode, and then control of multiple zones through the use of dampers. When these systems have temperature sensors in these zones, a layer of "smart" control is available so that users can set individual target temperatures per zone.

Since these zones have a target temp, current temp, and also support an ON/OFF type state, they are almost perfect fits for climate entities, except that ON doesn't exist in the HVAC_MODE.

In reality that HVAC_MODE is mirroring whatever the parent HVAC system is set to, but if an integration was to implement this, it would remove the ability to turn OFF a zone, without turning the entire HVAC system off.

I am proposing that **HVAC_MODE_AUTO** have its definition expanded to include where the HVAC mode has been decided by another system/process/entity, or to add a more generic **HVAC_MODE_ON** for systems that do not have a more specific HVAC mode.

I had originally thought **HVAC_MODE_FAN_ONLY** met this need, but it does not support setting the target temperature in platforms such as Google Assistant. In that case, Google do define an **ON** state, although its definition is intended to restore the previous state. https://developers.google.com/assistant/smarthome/traits/temperaturesetting

The specific example is my advantage_air integration, and PR https://github.com/home-assistant/core/pull/52816

I do not believe **HVAC_MODE_HEAT_COOL** is an appropriate alternative as it adds the complexity of temperature ranges.

**Comment by Noltari** [comment]
Hi @Bre77,

This is also the case for `airzone` integration, in which slave (child) zones can only be turned on/off and change their setpoint temperature, but the operation mode (`heat`, `cool`, etc.) can't be changed.
In this case, what I'm doing is exposing all supported HVAC modes, but if the user selects a different operation mode on a slave zone (vs what's configured on the parent/master zone) it will raise an exception:
https://github.com/home-assistant/core/blob/be05a749c57a7174839e75376a345bd184e97e0b/homeassistant/components/airzone/climate.py#L204-L207

However, this is a problem when there are limitations like in the homekit API:
https://github.com/home-assistant/core/issues/94665

**Comment by Noltari** [comment]
Hey @frenck, any chance we could add HVAC_MODE_ON mode or broaden the scope of HVAC_MODE_AUTO?

**Reply by frenck** [reply]
That has not been decided, nor has there been much support shown from other core members.

Could you please, for the future, withhold from pinging/pulling people into conversations? Thanks 👍

**Reply by Noltari** [reply]
> That has not been decided, nor has there been much support shown from other core members.

Ok, thanks for the info.

> Could you please, for the future, withhold from pinging/pulling people into conversations? Thanks 👍

Well, since this was opened back in 2021 and you have recently solved other climate discussions such as https://github.com/home-assistant/architecture/discussions/710#discussioncomment-6189015 I was just trying to give this a little push.
Excuse me if that annoyed you :(

**Reply by frenck** [reply]
> I was just trying to give this a little push.

I understand, don't worry 👍 It is just that with the size of the project, being pinged for everything and everyone isn't really productive :)

../Frenck

**Comment by gjohansson-ST** [comment]
I am struggling to understand the difference between `auto` and `on` in this case.
Why isn't `auto` a good pick?

**Reply by Noltari** [reply]
See https://developers.home-assistant.io/docs/core/entity/climate/#hvac-modes:
The device is set to a schedule, learned behavior, AI.
It doesn't cover the ON case for these devices, becase it's none of the 3 cases.

**Reply by gjohansson-ST** [reply]
What is then the `on` case?
Seems as you wrote that it has a target temp, `auto` would be a match?

**Reply by Bre77** [reply]
Not by the Home Assistant definition, which is why I raised this architecture decision.

I was forced to use HEAT_COOL, which isn't ideal since it doesn't actually reflect what's happening.

I agree AUTO is a good choice, which is why this decision asks for its definition to be broadened.
